File: 数据结构/散列表/128.最长连续序列.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int longestConsecutive = 0;
        // c++ set初始化
        unordered_set<int> numSet;
        for(int num : nums){
            // c++ set添加元素
            numSet.insert(num);
        }

        int result = 0;
        // c++ 遍历集合, 使用引用
        for(const int& num: numSet){
            // c++ 判断集合是否含有元素
            if(!numSet.count(num-1)){
                int curNum = num;
                int consecutive = 1;

                while(numSet.count(curNum+1)){
                    curNum += 1;
                    consecutive += 1;
                }

                result = max(result, consecutive);
            }
        }
        return result;
    }
};

File: 数据结构/散列表/128.最长连续序列.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if nums is None or len(nums) == 0 :
            return 0
        vis = set(nums)
        longest = 0
        for num in vis:
            if num - 1 in vis:
                # 为什么要加这一行?
                # 如果不加这行, 倒序的用例,时间复杂度会变成O(n^2)
                continue
            cur_num = num
            consecutive = 1
            while cur_num + 1 in vis:
                consecutive = consecutive + 1
                cur_num = cur_num + 1
            
            longest = max(longest, consecutive)
        return longest
            
            
             